text/0000-payment-proofs.md

```diff
@@ -0,0 +1,101 @@
+- Title: payment-proofs
+- Authors: [David Burkett](mailto:davidburkett38@gmail.com)
+
+---
+
+# Summary
+[summary]: #summary
+
+Support generating and validating payment proofs for sender-initiated (ie. non-invoice) transactions. 
+
+# Motivation
+[motivation]: #motivation
+
+Bitcoin and other cryptocurrencies with transparent protocol-level addressing and immutable, unpruneable blockchains can prove sender, receiver, and amounts of payments simply by pointing to the transaction in the blockchain.
+Grin's privacy and scalability means users no longer have this ability. This prevents some merchants from accepting Grin due to the high possibility of payment disputes that are unresolvable in the same way they are for transparent coins.
+
+This RFC proposes a change to the transaction building process where payers can require payees to create a "proof" they've received a payment before the payer finalizes and broadcasts the transaction.
+
+# Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+From an end-user perspective, payers can require payees to prove receipt of funds as part of the transacting process.
+Payers can then use these "proofs", along with information from the blockchain, to resolve payment disputes and prove funds were sent to the correct payee, and the transaction was confirmed on the blockchain.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Slate changes
+
+A new (optional) structure (`payment_info`) will be added to transaction slates, along with a version increase. The `payment_info` structure will contain:
+
+* `message` - Optional message to include in the payment proof. Limited to 100(?) characters.
+* `sender_address` - Optional ed25519 public key generated by the sender.
+* `receiver_address` - An ed25519 public key for the receiver, typically the public key of the user's v3 onion address.
+* `receiver_signature` - A signature of the sender_address, received amount, and kernel commitment that validates against the `receiver_address`. 
+
+## Generating proofs
+
+Receipt confirmations (`receiver_signature`s) will be generated by the payee by providing an ED25519 signature of: `(amount || kernel_commitment || sender_address || message)`, using the private key of the `receiver_address`.
```

Comment:
Should this not be `(amount || kernel_commitment || **receiver_address** || message)`?

Reply:
No, it's signed using the key of the receiver_address, so no need to include it in the signature message.

Reply:
Right, was just a bit thrown because the sender address is optional. What’s the details of the format if the sender address doesn’t exist?

Reply:
If sender_address or message don't exist, just drop them from the signature message. So `(amount || kernel_commitment || message)` if no sender address, `(amount || kernel_commitment || sender_address)` if no message, or `(amount || kernel_commitment)` if neither.

Reply:
It think it would be better to keep this consistent, but since we're all in agreement that the sender address should be mandatory, the sig message can just be defined as (amount || kernel_commit || sender_address || message) where message is the empty string  "" if it doesn't exist (which amounts to the same thing anyhow)